### ISSUE TITLE
Problem in Smooch.init

### ISSUE THREAD

**Issue by ivanespitiac** [zendesk/sunshine-conversations-android#135]
**For support questions, please visit [our help page](https://smooch.io/help/)**.

This repository's issues are reserved for bug reports only.

 **I'm submitting a ...**
  - [

When use smooch.init(app, settings), show this response error "An unexpected error occurred during initialization. Retrying in 96 seconds... " but never make init, it's an infinity loop, with Status response: 200.

 ] bug report
  - [ ] support request => Please do not submit support request here, see note at the top of this template.

 **What is the current behavior?**

When use smooch.init(app, settings), show this response error "An unexpected error occurred during initialization. Retrying in 96 seconds... " but never make init, it's an infinity loop, with Status response: 200.

 **Please provide the steps to reproduce and, if possible, a minimal demo of the problem**. Ideally a sample project

val compactJws = Jwts.builder()
                    .claim("scope", "######")
                    .claim("userId", "######")
                    .setSubject("######")
                    .signWith("############")
                    .setHeaderParam("kid", "######")
                    .setHeaderParam("alg", "######")
                    .setHeaderParam("typ", "######")
                    .compact()

            val settings = Settings("######")
            settings.fileProviderAuthorities = "######"
            settings.isFirebaseCloudMessagingAutoRegistrationEnabled = false
            settings.authenticationDelegate = AuthenticationDelegate { _, p1 -> p1?.updateToken(compactJws) }

            Smooch.init(app, settings) { response ->
val logMessage = "Response Init smooch ${response.error}  Data: ${response.data} Status: ${response.status}"
                logMessage.log()
            }

 **What is the expected behavior?**

Good init

 **Please tell us about your environment:**
- SDK version:

    compileSdkVersion 28
    buildToolsVersion = '28.0.3'

- OS version

Android 6.0, 8.0

 **Other information** (e.g. detailed explanation, stack traces, related issues, suggestions how to fix, links for us to have context, eg. Stack Overflow, etc)

**Comment by wmora** [comment]
I couldn't reproduce the issue. This is the code I ran in my Application class:

```
val compactJws = Jwts.builder()
            .claim("scope", "######")
            .claim("userId", "######")
            .setHeaderParam("kid", " "######")
            .signWith(
                Keys.hmacShaKeyFor("######".toByteArray()),
                SignatureAlgorithm.HS256)
            .compact()

        val settings = Settings("######")
        settings.isFirebaseCloudMessagingAutoRegistrationEnabled = false
        settings.authenticationDelegate = AuthenticationDelegate { _, p1 -> p1?.updateToken(compactJws) }

        Smooch.init(this, settings) { response ->
            val logMessage = "Response Init smooch ${response.error} Data: ${response.data} Status: ${response.status}"
            println(logMessage)
        }
```

Also it's strongly discouraged to sign JWTs in your application (i.e. in Kotlin), as it implies distributing your app secret key with your mobile application, essentially granting all of your users access to your entire Smooch app. JWTs should be signed on your server and passed down to the client device.

**Comment by ivanespitiac** [comment]
the problem appear when make release build, using proguard optimize or R8.

**Comment by DataWearsAHood** [comment]
Hello Ivan,

Looks like a few Proguard fixed have been shipped since this issue was first reported (v7.0.0, v7.0.3).
Please reopen this issue if you are still experiencing this crash with the latest release.
